### PR TITLE
Add Indiana EITC and in_income_tax variable

### LINKED ISSUE #2808: Add Indiana EITC and elderly credit as refundable tax credits
Plus rationalize the variable and parameter tree structures.

### PR DESCRIPTION
**Pull request by martinholmer**
Fixes #2808.

The changes so far cause there to be no integration test failures in the 600,000 tests in the `e21` through `j21` samples described in issue #993.  But because the decoupled EITC code is not yet specified, there are 19,159 failures (out of 100,000) in the `p21` sample.  Also, there are many itemized deduction failures in the `k21` sample.

All of the above mentioned differences have been eliminated after a series of subsequent commits that added the decoupled EITC logic.

### Summary
:memo::truck::sparkles:

This pull request implements the Indiana state income tax system, including the adjusted gross income tax, the earned income tax credit, and the refundable credits. It also renames and reorganizes some files and parameters for clarity and consistency. It adds a changelog entry file and several unit tests to verify the tax calculations.

> _`in_eitc` added_
> _parameters and tests updated_
> _autumn of tax code_

### Walkthrough
*  Add a changelog entry file for the pull request ([link](https://github.com/PolicyEngine/policyengine-us/pull/2815/files?diff=unified&w=0#diff-5885042762833a4aef51405a4036067e4e6aefa7a5091a2987d1d9d1fc2647e8R1))
*  Simplify the file structure and parameter name for the Indiana AGI tax rate ([link](https://github.com/PolicyEngine/policyengine-us/pull/2815/files?diff=unified&w=0#diff-e41ab910986aa44f1c524905c79a92c3bfc4a06de0a20fa26884673a29a8ca3fL1-R11))
*  Introduce new parameters for the Indiana EITC, including the decoupling status, the match rate, and the refundable credits ([link](https://github.com/PolicyEngine/policyengine-us/pull/2815/files?diff=unified&w=0#diff-60f2b461c8923d429baa6fa0ec3407a7114f8b0938db2792d8045adc685f7424R1-R10), [link](https://github.com/PolicyEngine/policyengine-us/pull/2815/files?diff=unified&w=0#diff-61ecc2eab89f17f887aaef80ad0e2a115fa8636bc97ef46402c1d4c9243a6564R1-R11), [link](https://github.com/PolicyEngine/policyengine-us/pull/2815/files?diff=unified&w=0#diff-5dfab5493a01d6499a795ab1cad60df689ab6a7b42de31735ac1730c02c0f975R1-R11))
*  Rename the test files for the Indiana income tax system to match the variable names ([link](https://github.com/PolicyEngine/policyengine-us/pull/2815/files?diff=unified&w=0#diff-301f5986655041d89f196f1dc3cb071248bdc272d859395a9227751e2d3da638), [link](https://github.com/PolicyEngine/policyengine-us/pull/2815/files?diff=unified&w=0#diff-b1c216397a1cfc23f2b27a82ba6ed508ca536c3114fc1d0d7df037ec1d1bf77e), [link](https://github.com/PolicyEngine/policyengine-us/pull/2815/files?diff=unified&w=0#diff-4c8db4cdcb5cc5f811b5509c9cbcd77a4be55c3ae200dea44c9d27737abb7847),  [link](https://github.com/PolicyEngine/policyengine-us/pull/2815/files?diff=unified&w=0#diff-10de92fe43ffb9726dff18991ca524eb422d3ef44171229e87a2652b5cb4c701), [link](https://github.com/PolicyEngine/policyengine-us/pull/2815/files?diff=unified&w=0#diff-f9726b6a74ec0fba70e29c2985c2adc567398e00ea72279f485af1e190e0d297R1-R19), [link](https://github.com/PolicyEngine/policyengine-us/pull/2815/files?diff=unified&w=0#diff-8b0aeb118bf499faac393fd956b66d968e363bba931dd1551b2d276bbd835d63R1-R77), [link](https://github.com/PolicyEngine/policyengine-us/pull/2815/files?diff=unified&w=0#diff-e3b367ad9e55e22dd9f94432b5a21fa96884d94c150722d55cd0f24ded80e090), [link](https://github.com/PolicyEngine/policyengine-us/pull/2815/files?diff=unified&w=0#diff-ad78885e2b8636c7174b6db55eb7386513979ae6725b4951180e89b6ef354928), [link](https://github.com/PolicyEngine/policyengine-us/pull/2815/files?diff=unified&w=0#diff-f0f892d252f7fb014cf1d3f3965b858266a6aee61e46559d47dc512bf2dab354), [link](https://github.com/PolicyEngine/policyengine-us/pull/2815/files?diff=unified&w=0#diff-f0479a47cf8eccfbc7f92004e037bf5febe580d759bcb58abf1684b68e738146), [link](https://github.com/PolicyEngine/policyengine-us/pull/2815/files?diff=unified&w=0#diff-55c019d3c0619f087ba97784fe5af304fb349fa37507c013c98f9af2040831e8), [link](https://github.com/PolicyEngine/policyengine-us/pull/2815/files?diff=unified&w=0#diff-b6a98d0dc94f6aa38055e50014846ae3f150ec884f6dcd6441bd665646aede06), [link](https://github.com/PolicyEngine/policyengine-us/pull/2815/files?diff=unified&w=0#diff-44caf13015d62f9ec28ce8556f1ede0adc7dbad4b311dfeb01bcd74e4bcb2096))
*  Add new test cases for the Indiana AGI tax, the EITC eligibility and amount, and the qualifying dependent child indicator ([link](https://github.com/PolicyEngine/policyengine-us/pull/2815/files?diff=unified&w=0#diff-241e139449984f84471c9e45e118edaa586eb25cea5b0aebd4039ec67bc14e90R16-R23), [link](https://github.com/PolicyEngine/policyengine-us/pull/2815/files?diff=unified&w=0#diff-f9726b6a74ec0fba70e29c2985c2adc567398e00ea72279f485af1e190e0d297R1-R19), [link](https://github.com/PolicyEngine/policyengine-us/pull/2815/files?diff=unified&w=0#diff-8b0aeb118bf499faac393fd956b66d968e363bba931dd1551b2d276bbd835d63R1-R77), [link](https://github.com/PolicyEngine/policyengine-us/pull/2815/files?diff=unified&w=0#diff-f0f892d252f7fb014cf1d3f3965b858266a6aee61e46559d47dc512bf2dab354))
*  Delete the unused README file for the Indiana AGI tax ([link](https://github.com/PolicyEngine/policyengine-us/pull/2815/files?diff=unified&w=0#diff-cd39b91bb83daf4c7a440e1deea59ab77aa08134b1740c116b59f0c62b0e9663))

### CODE REVIEW

File: policyengine_us/parameters/gov/states/in/tax/income/agi_rate.yaml [hunk 1]
```diff
@@ -1,13 +1,11 @@
-description: IN adjusted gross income tax rate.
-values:
-  2014-01-01: 0.034 # "For taxable years beginning before January 1, 2015"
-  2015-01-01: 0.033
-  2017-01-01: 0.0323
+description: Indiana adjusted gross income tax rate.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/in/tax/income/agi_rate.yaml:1]:
```suggestion
description: Indiana taxes adjusted gross income at this rate.
```

File: policyengine_us/parameters/gov/states/in/tax/income/agi_rate.yaml [hunk 2]
```diff
@@ -1,13 +1,11 @@
...
 metadata:
   period: year
   unit: /1
-  name: in_agi_tax_rate
-  label: IN AGI tax rate
   reference:
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/in/tax/income/agi_rate.yaml:5]:
```suggestion
  label: Indiana AGI tax rate
```

File: policyengine_us/parameters/gov/states/in/tax/income/credits/earned_income/childless/max_age.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+description: Indiana has a state EITC that used this maximum age for childless eligiblity.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/in/tax/income/credits/earned_income/childless/max_age.yaml:1]:
```suggestion
description: Indiana limits its earned income tax credit to filers this age or below if they have no children.
```

File: policyengine_us/parameters/gov/states/in/tax/income/credits/earned_income/childless/max_age.yaml [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+metadata:
+  unit: float
+  label: IN EITC maximum age for childless eligibility
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/in/tax/income/credits/earned_income/childless/max_age.yaml:4]:
```suggestion
  label: Indiana EITC maximum age for childless eligibility
```

File: policyengine_us/variables/gov/states/in/tax/income/credits/elderly_tax_credit/in_unified_elderly_tax_credit.py [hunk 5]
```diff
@@ -7,31 +7,29 @@ class in_unified_elderly_tax_credit(Variable):
...
-        eligible = head_eligible | spouse_eligible
-        married_amount = where(
-            both_eligible,
-            p.amount.married.two_aged.calc(total_income),
-            p.amount.married.one_aged.calc(total_income),
+        aged_spouse = (spouse_age >= p.unified_elderly.min_age).astype(int)
+        aged_count = aged_head + aged_spouse
+        return select(
+            [
+                aged_count == 0,
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/in/tax/income/credits/elderly_tax_credit/in_unified_elderly_tax_credit.py:26]: `default` is more performant https://github.com/PolicyEngine/policyengine-us/issues/1176
```suggestion
```

File: policyengine_us/variables/gov/states/in/tax/income/credits/elderly_tax_credit/in_unified_elderly_tax_credit.py [hunk 6]
```diff
@@ -7,31 +7,29 @@ class in_unified_elderly_tax_credit(Variable):
...
+                aged_count == 1,
+                aged_count == 2,
+            ],
+            [
+                0,
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/in/tax/income/credits/elderly_tax_credit/in_unified_elderly_tax_credit.py:31]:
```suggestion
```

File: policyengine_us/variables/gov/states/in/tax/income/credits/elderly_tax_credit/in_unified_elderly_tax_credit.py [hunk 7]
```diff
@@ -7,31 +7,29 @@ class in_unified_elderly_tax_credit(Variable):
...
+                p.unified_elderly.amount.one_aged.calc(federal_agi),
+                p.unified_elderly.amount.two_aged.calc(federal_agi),
+            ],
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/in/tax/income/credits/elderly_tax_credit/in_unified_elderly_tax_credit.py:34]:
```suggestion
            ],
            default=0
```

File: policyengine_us/variables/gov/states/in/tax/income/credits/in_refundable_credits.py [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+
+class in_refundable_credits(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Indiana refundable income tax credits"
+    unit = USD
+    definition_period = YEAR
+    reference = "https://iga.in.gov/laws/2021/ic/titles/6#6-3.1"
+    defined_for = StateCode.IN
+
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/in/tax/income/credits/in_refundable_credits.py:12]: Replacing `formula`, unless the `in` causes this to fail?
```suggestion
    adds = "gov.states.in.tax.income.credits.refundable"
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]
Could you apply the suggested label/description phrasings to other parameters as well?

Merging this before fixing https://github.com/PolicyEngine/policyengine-api/issues/230 will cause existing household links in the app to break, because it (appropriately) renames some variables. So we may either want to wait on merging this until we fix that, or leave placeholder variables for those that this PR removes. @nikhilwoodruff @anth-volk

File: policyengine_us/variables/gov/states/in/tax/income/credits/in_refundable_credits.py (hunk 8 above)
Comment (martinholmer) [reply policyengine_us/variables/gov/states/in/tax/income/credits/in_refundable_credits.py:12]: As I remember, it did fail, hence the use of the formula.
To do what you are suggesting, you would have to get @nikhilwoodruff to enhance PolicyEngine-Core.

File: policyengine_us/variables/gov/states/in/tax/income/credits/elderly_tax_credit/in_unified_elderly_tax_credit.py (hunk 7 above)
Comment (martinholmer) [reply policyengine_us/variables/gov/states/in/tax/income/credits/elderly_tax_credit/in_unified_elderly_tax_credit.py:34]: Used default=0 in select statement.

File: policyengine_us/parameters/gov/states/in/tax/income/credits/earned_income/childless/max_age.yaml (hunk 4 above)
Comment (martinholmer) [reply policyengine_us/parameters/gov/states/in/tax/income/credits/earned_income/childless/max_age.yaml:4]: Replaced IN with Indiana in all six childless parameter files.

File: policyengine_us/parameters/gov/states/in/tax/income/credits/earned_income/childless/max_age.yaml (hunk 3 above)
Comment (martinholmer) [reply policyengine_us/parameters/gov/states/in/tax/income/credits/earned_income/childless/max_age.yaml:1]: Changed description in both the childless/min_age.yaml and childless/max_age.yaml file.

File: policyengine_us/parameters/gov/states/in/tax/income/agi_rate.yaml (hunk 2 above)
Comment (martinholmer) [reply policyengine_us/parameters/gov/states/in/tax/income/agi_rate.yaml:5]: Added label: to agi_rate.yaml parameter file.

File: policyengine_us/parameters/gov/states/in/tax/income/agi_rate.yaml (hunk 1 above)
Comment (martinholmer) [reply policyengine_us/parameters/gov/states/in/tax/income/agi_rate.yaml:1]: Done.

File: policyengine_us/variables/gov/states/in/tax/income/credits/in_refundable_credits.py (hunk 8 above)
Comment (MaxGhenis) [reply policyengine_us/variables/gov/states/in/tax/income/credits/in_refundable_credits.py:12]: Did you try `adds = "gov.states['in'].tax.income.credits.refundable"`?

Comment (MaxGhenis) [reply policyengine_us/variables/gov/states/in/tax/income/credits/in_refundable_credits.py:12]: Also I filed https://github.com/PolicyEngine/policyengine-core/issues/112 for a core workaround

**Comment by martinholmer** [comment]
@MaxGhenis said in the discussion of Indiana PR #2815:
> Merging this before fixing https://github.com/PolicyEngine/policyengine-api/issues/230 will cause existing household links in the app to break, because it (appropriately) renames some variables.

I don't understand why Indiana was included in the WebApp when it was seriously incomplete.

**Comment by MaxGhenis** [comment]
Indiana is not in the web app. The issue occurs any time we remove a variable from the Python package, whether it's hooked up to the net income tree or not. We may be able to avoid it more by adopting feature branches, but really we need to fix it for all cases.

**Comment by anth-volk** [comment]
> Could you apply the suggested label/description phrasings to other parameters as well?
> 
> Merging this before fixing [PolicyEngine/policyengine-api#230](https://github.com/PolicyEngine/policyengine-api/issues/230) will cause existing household links in the app to break, because it (appropriately) renames some variables. So we may either want to wait on merging this until we fix that, or leave placeholder variables for those that this PR removes. @nikhilwoodruff @anth-volk

I had been waiting to address [API #230](https://github.com/PolicyEngine/policyengine-api/issues/230) until the incorporation of a different PR to allow for better back-end testing, as I had been hoping to build against these tests. That said, I think we may want to reopen the discussion on how to best solve [API #230](https://github.com/PolicyEngine/policyengine-api/issues/230), as I wonder if there are stronger solutions we might be missing.

**Comment by MaxGhenis** [comment]
I just got some additional interest in Indiana policy, so I'm going to merge this and sacrifice the historical household links. Anthony and Nikhil, appreciate any priority you're able to give to https://github.com/PolicyEngine/policyengine-api/issues/230.

**Pull request merged**